boa_engine/src/vm/code_block.rs

```diff
@@ -129,7 +134,10 @@ pub struct CodeBlock {
     pub(crate) functions: Box<[Gc<Self>]>,
 
     /// Compile time environments in this function.
-    pub(crate) compile_environments: Box<[Gc<GcRefCell<CompileTimeEnvironment>>]>,
+    ///
+    // Safety: Nothing in CompileTimeEnvironment needs tracing, so this is safe.
```

Comment:
Maybe we should add a comment in `CompileTimeEnvironment` so that we don't forget and add a Gc there in the future.

Reply:
I think this should already be tracked in #3034, right?